modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java

```diff
@@ -652,11 +655,26 @@ private void _sendUserNotificationEvents(ObjectDefinition objectDefinition)
 			objectDefinition.getCompanyId(), RoleConstants.ADMINISTRATOR);
 
 		for (long userId : _userLocalService.getRoleUserIds(role.getRoleId())) {
-			int count =
-				_userNotificationEventLocalService.
-					getUserNotificationEventsCount(
-						userId, portletId, timestamp, true,
-						"%\"exceedsObjectEntryLimit\":%");
+			List<UserNotificationEvent> userNotificationEventList =
+				_userNotificationEventLocalService.getUserNotificationEvents(
+					userId, portletId, timestamp, true);
+
+			int count = 0;
+
+			for (UserNotificationEvent userNotificationEvent :
+					userNotificationEventList) {
+
+				if (count > 0) {
```

Comment:
What do you think to change this whole logic to the code below?

```
		for (long userId : _userLocalService.getRoleUserIds(role.getRoleId())) {
			List<UserNotificationEvent> userNotificationEvents =
				_userNotificationEventLocalService.getUserNotificationEvents(
					userId, portletId, timestamp, true);

			boolean notificationReceived = false;

			for (UserNotificationEvent userNotificationEvent :
					userNotificationEvents) {

				JSONObject jsonObject = _jsonFactory.createJSONObject(
					userNotificationEvent.getPayload());

				if (jsonObject.has("exceedsObjectEntryLimit")) {
					notificationReceived = true;
					break;
				}
			}

			if (!notificationReceived) {
				userIds.add(userId);
			}
		}
```

Reply:
I thought that using "break" could be a problem. So I used other way of doing this loop. If it is not a problem I can use it.

Reply:
It's not a problem, you can use.